// @nativescript/angular/common.js
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("@angular/common");
var core_1 = require("@angular/core");
var dialogs_1 = require("./directives/dialogs");
var platform_providers_1 = require("./platform-providers");
var list_view_comp_1 = require("./directives/list-view-comp");
var templated_items_comp_1 = require("./directives/templated-items-comp");
var tab_view_1 = require("./directives/tab-view");
var action_bar_1 = require("./directives/action-bar");
var platform_filters_1 = require("./directives/platform-filters");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/common');
var ɵngcc2 = require('./directives/list-view-comp');
var ɵngcc3 = require('./directives/templated-items-comp');
var ɵngcc4 = require('./directives/tab-view');
var ɵngcc5 = require('./directives/action-bar');
var ɵngcc6 = require('./directives/platform-filters');
var NativeScriptCommonModule = /** @class */ (function () {
    function NativeScriptCommonModule() {
    }
NativeScriptCommonModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NativeScriptCommonModule });
NativeScriptCommonModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NativeScriptCommonModule_Factory(t) { return new (t || NativeScriptCommonModule)(); }, providers: [
        dialogs_1.ModalDialogService,
        platform_providers_1.defaultDeviceProvider,
        platform_providers_1.defaultFrameProvider,
        platform_providers_1.defaultPageProvider,
    ], imports: [[common_1.CommonModule], ɵngcc1.CommonModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NativeScriptCommonModule, { declarations: [ɵngcc2.ListViewComponent, ɵngcc3.TemplateKeyDirective, ɵngcc4.TabViewDirective, ɵngcc4.TabViewItemDirective, ɵngcc5.ActionBarComponent, ɵngcc5.ActionBarScope, ɵngcc5.ActionItemDirective, ɵngcc5.NavigationButtonDirective, ɵngcc6.AndroidFilterComponent, ɵngcc6.IosFilterComponent], imports: [ɵngcc1.CommonModule], exports: [ɵngcc1.CommonModule, ɵngcc2.ListViewComponent, ɵngcc3.TemplateKeyDirective, ɵngcc4.TabViewDirective, ɵngcc4.TabViewItemDirective, ɵngcc5.ActionBarComponent, ɵngcc5.ActionBarScope, ɵngcc5.ActionItemDirective, ɵngcc5.NavigationButtonDirective, ɵngcc6.AndroidFilterComponent, ɵngcc6.IosFilterComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NativeScriptCommonModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [
                    list_view_comp_1.ListViewComponent,
                    templated_items_comp_1.TemplateKeyDirective,
                    tab_view_1.TabViewDirective,
                    tab_view_1.TabViewItemDirective,
                    action_bar_1.ActionBarComponent,
                    action_bar_1.ActionBarScope,
                    action_bar_1.ActionItemDirective,
                    action_bar_1.NavigationButtonDirective,
                    platform_filters_1.AndroidFilterComponent,
                    platform_filters_1.IosFilterComponent,
                ],
                providers: [
                    dialogs_1.ModalDialogService,
                    platform_providers_1.defaultDeviceProvider,
                    platform_providers_1.defaultFrameProvider,
                    platform_providers_1.defaultPageProvider,
                ],
                imports: [common_1.CommonModule],
                exports: [
                    common_1.CommonModule,
                    list_view_comp_1.ListViewComponent,
                    templated_items_comp_1.TemplateKeyDirective,
                    tab_view_1.TabViewDirective,
                    tab_view_1.TabViewItemDirective,
                    action_bar_1.ActionBarComponent,
                    action_bar_1.ActionBarScope,
                    action_bar_1.ActionItemDirective,
                    action_bar_1.NavigationButtonDirective,
                    platform_filters_1.AndroidFilterComponent,
                    platform_filters_1.IosFilterComponent,
                ],
                schemas: [core_1.NO_ERRORS_SCHEMA]
            }]
    }], function () { return []; }, null); })();
    return NativeScriptCommonModule;
}());
exports.NativeScriptCommonModule = NativeScriptCommonModule;
__export(require("./directives"));
