// @nativescript/angular/nativescript.module.js
Object.defineProperty(exports, "__esModule", { value: true });
require("@nativescript/core/globals");
// Require application early to work around a circular import
require("@nativescript/core/application");
require("./polyfills/array");
require("./polyfills/console");
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var common_2 = require("./common");
var renderer_1 = require("./renderer");
var detached_loader_1 = require("./common/detached-loader");
exports.DetachedLoader = detached_loader_1.DetachedLoader;
var utils_1 = require("./common/utils");
var platform_providers_1 = require("./platform-providers");
function errorHandlerFactory() {
    return new core_1.ErrorHandler();
}
exports.errorHandlerFactory = errorHandlerFactory;
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./common');
var ɵngcc2 = require('./common/detached-loader');
var NativeScriptModule = /** @class */ (function () {
    function NativeScriptModule(parentModule) {
        // Prevents NativeScriptModule from getting imported multiple times
        utils_1.throwIfAlreadyLoaded(parentModule, "NativeScriptModule");
    }
    NativeScriptModule = __decorate([ __param(0, core_1.Optional()), __param(0, core_1.SkipSelf()),
        __metadata("design:paramtypes", [NativeScriptModule])
    ], NativeScriptModule);
NativeScriptModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NativeScriptModule });
NativeScriptModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NativeScriptModule_Factory(t) { return new (t || NativeScriptModule)(ɵngcc0.ɵɵinject(NativeScriptModule, 12)); }, providers: [
        platform_providers_1.FrameService,
        platform_providers_1.PageService,
        renderer_1.NativeScriptRendererFactory,
        core_1.SystemJsNgModuleLoader,
        { provide: core_1.ɵINJECTOR_SCOPE, useValue: "root" },
        { provide: core_1.ErrorHandler, useFactory: errorHandlerFactory },
        { provide: core_1.RendererFactory2, useExisting: renderer_1.NativeScriptRendererFactory },
        { provide: common_1.ViewportScroller, useClass: common_1.ɵNullViewportScroller },
    ], imports: [[
            core_1.ApplicationModule,
            common_2.NativeScriptCommonModule,
        ], ɵngcc0.ApplicationModule, ɵngcc1.NativeScriptCommonModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NativeScriptModule, { declarations: [ɵngcc2.DetachedLoader], imports: [ɵngcc0.ApplicationModule, ɵngcc1.NativeScriptCommonModule], exports: [ɵngcc0.ApplicationModule, ɵngcc1.NativeScriptCommonModule, ɵngcc2.DetachedLoader] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NativeScriptModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [
                    detached_loader_1.DetachedLoader,
                ],
                providers: [
                    platform_providers_1.FrameService,
                    platform_providers_1.PageService,
                    renderer_1.NativeScriptRendererFactory,
                    core_1.SystemJsNgModuleLoader,
                    { provide: core_1.ɵINJECTOR_SCOPE, useValue: "root" },
                    { provide: core_1.ErrorHandler, useFactory: errorHandlerFactory },
                    { provide: core_1.RendererFactory2, useExisting: renderer_1.NativeScriptRendererFactory },
                    { provide: common_1.ViewportScroller, useClass: common_1.ɵNullViewportScroller },
                ],
                entryComponents: [
                    detached_loader_1.DetachedLoader,
                ],
                imports: [
                    core_1.ApplicationModule,
                    common_2.NativeScriptCommonModule,
                ],
                exports: [
                    core_1.ApplicationModule,
                    common_2.NativeScriptCommonModule,
                    detached_loader_1.DetachedLoader,
                ],
                schemas: [core_1.NO_ERRORS_SCHEMA]
            }]
    }], function () { return [{ type: NativeScriptModule, decorators: [{
                type: core_1.Optional
            }, {
                type: core_1.SkipSelf
            }] }]; }, null); })();
    return NativeScriptModule;
}());
exports.NativeScriptModule = NativeScriptModule;
