// @nativescript/angular/http-client/http-client.module.js
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var http_1 = require("@angular/common/http");
var ns_file_system_1 = require("../file-system/ns-file-system");
var ns_http_backend_1 = require("./ns-http-backend");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/common/http');
var NativeScriptHttpClientModule = /** @class */ (function () {
    function NativeScriptHttpClientModule() {
    }
NativeScriptHttpClientModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NativeScriptHttpClientModule });
NativeScriptHttpClientModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NativeScriptHttpClientModule_Factory(t) { return new (t || NativeScriptHttpClientModule)(); }, providers: [
        ns_file_system_1.NSFileSystem,
        ns_http_backend_1.NsHttpBackEnd,
        { provide: http_1.HttpBackend, useExisting: ns_http_backend_1.NsHttpBackEnd },
    ], imports: [[
            http_1.HttpClientModule,
        ], ɵngcc1.HttpClientModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NativeScriptHttpClientModule, { imports: [ɵngcc1.HttpClientModule], exports: [ɵngcc1.HttpClientModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NativeScriptHttpClientModule, [{
        type: core_1.NgModule,
        args: [{
                providers: [
                    ns_file_system_1.NSFileSystem,
                    ns_http_backend_1.NsHttpBackEnd,
                    { provide: http_1.HttpBackend, useExisting: ns_http_backend_1.NsHttpBackEnd },
                ],
                imports: [
                    http_1.HttpClientModule,
                ],
                exports: [
                    http_1.HttpClientModule,
                ]
            }]
    }], function () { return []; }, null); })();
    return NativeScriptHttpClientModule;
}());
exports.NativeScriptHttpClientModule = NativeScriptHttpClientModule;
