// @nativescript/angular/forms/forms.module.js
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var value_accessors_1 = require("./value-accessors");
exports.TextValueAccessor = value_accessors_1.TextValueAccessor;
exports.CheckedValueAccessor = value_accessors_1.CheckedValueAccessor;
exports.DateValueAccessor = value_accessors_1.DateValueAccessor;
exports.TimeValueAccessor = value_accessors_1.TimeValueAccessor;
exports.NumberValueAccessor = value_accessors_1.NumberValueAccessor;
exports.SelectedIndexValueAccessor = value_accessors_1.SelectedIndexValueAccessor;
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/forms');
var ɵngcc2 = require('./value-accessors/text-value-accessor');
var ɵngcc3 = require('./value-accessors/checked-value-accessor');
var ɵngcc4 = require('./value-accessors/date-value-accessor');
var ɵngcc5 = require('./value-accessors/time-value-accessor');
var ɵngcc6 = require('./value-accessors/selectedIndex-value-accessor');
var ɵngcc7 = require('./value-accessors/number-value-accessor');
var NativeScriptFormsModule = /** @class */ (function () {
    function NativeScriptFormsModule() {
    }
NativeScriptFormsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NativeScriptFormsModule });
NativeScriptFormsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NativeScriptFormsModule_Factory(t) { return new (t || NativeScriptFormsModule)(); }, providers: [], imports: [[forms_1.FormsModule], ɵngcc1.FormsModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NativeScriptFormsModule, { declarations: [ɵngcc2.TextValueAccessor, ɵngcc3.CheckedValueAccessor, ɵngcc4.DateValueAccessor, ɵngcc5.TimeValueAccessor, ɵngcc6.SelectedIndexValueAccessor, ɵngcc7.NumberValueAccessor], imports: [ɵngcc1.FormsModule], exports: [ɵngcc1.FormsModule, ɵngcc2.TextValueAccessor, ɵngcc3.CheckedValueAccessor, ɵngcc4.DateValueAccessor, ɵngcc5.TimeValueAccessor, ɵngcc6.SelectedIndexValueAccessor, ɵngcc7.NumberValueAccessor] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NativeScriptFormsModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [
                    value_accessors_1.TextValueAccessor,
                    value_accessors_1.CheckedValueAccessor,
                    value_accessors_1.DateValueAccessor,
                    value_accessors_1.TimeValueAccessor,
                    value_accessors_1.SelectedIndexValueAccessor,
                    value_accessors_1.NumberValueAccessor,
                ],
                providers: [],
                imports: [forms_1.FormsModule],
                exports: [
                    forms_1.FormsModule,
                    value_accessors_1.TextValueAccessor,
                    value_accessors_1.CheckedValueAccessor,
                    value_accessors_1.DateValueAccessor,
                    value_accessors_1.TimeValueAccessor,
                    value_accessors_1.SelectedIndexValueAccessor,
                    value_accessors_1.NumberValueAccessor,
                ]
            }]
    }], function () { return []; }, null); })();
    return NativeScriptFormsModule;
}());
exports.NativeScriptFormsModule = NativeScriptFormsModule;
