// @nativescript/angular/router/ns-empty-outlet.component.js
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var page_1 = require("@nativescript/core/ui/page");
var page_router_outlet_1 = require("./page-router-outlet");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@nativescript/core/ui/page');
var ɵngcc2 = require('./page-router-outlet');
var NSEmptyOutletComponent = /** @class */ (function () {
    function NSEmptyOutletComponent(page) {
        var _this = this;
        this.page = page;
        if (this.page) {
            this.page.actionBarHidden = true;
            this.page.on("loaded", function () {
                if (_this.pageRouterOutlet && _this.page.frame) {
                    _this.pageRouterOutlet.setActionBarVisibility(_this.page.frame.actionBarVisibility);
                }
            });
        }
    }
    __decorate([
        core_1.ViewChild(page_router_outlet_1.PageRouterOutlet, { read: page_router_outlet_1.PageRouterOutlet, static: false }),
        __metadata("design:type", page_router_outlet_1.PageRouterOutlet)
    ], NSEmptyOutletComponent.prototype, "pageRouterOutlet", void 0);
    NSEmptyOutletComponent = __decorate([ __metadata("design:paramtypes", [page_1.Page])
    ], NSEmptyOutletComponent);
NSEmptyOutletComponent.ɵfac = function NSEmptyOutletComponent_Factory(t) { return new (t || NSEmptyOutletComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Page)); };
NSEmptyOutletComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NSEmptyOutletComponent, selectors: [["ns-empty-outlet"]], viewQuery: function NSEmptyOutletComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(page_router_outlet_1.PageRouterOutlet, true, page_router_outlet_1.PageRouterOutlet);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.pageRouterOutlet = _t.first);
    } }, decls: 1, vars: 0, consts: [["isEmptyOutlet", "true"]], template: function NSEmptyOutletComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "page-router-outlet", 0);
    } }, directives: [ɵngcc2.PageRouterOutlet], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NSEmptyOutletComponent, [{
        type: core_1.Component,
        args: [{
                // tslint:disable-next-line:component-selector
                selector: "ns-empty-outlet",
                template: "<page-router-outlet isEmptyOutlet='true'></page-router-outlet>"
            }]
    }], function () { return [{ type: ɵngcc1.Page }]; }, { pageRouterOutlet: [{
            type: core_1.ViewChild,
            args: [page_router_outlet_1.PageRouterOutlet, { read: page_router_outlet_1.PageRouterOutlet, static: false }]
        }] }); })();
    return NSEmptyOutletComponent;
}());
exports.NSEmptyOutletComponent = NSEmptyOutletComponent;
