// @nativescript/angular/directives/action-bar.js
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var action_bar_1 = require("@nativescript/core/ui/action-bar");
var page_1 = require("@nativescript/core/ui/page");
var lang_facade_1 = require("../lang-facade");
var element_registry_1 = require("../element-registry");
function isActionItem(view) {
    return view instanceof action_bar_1.ActionItem;
}
exports.isActionItem = isActionItem;
function isNavigationButton(view) {
    return view instanceof action_bar_1.NavigationButton;
}
exports.isNavigationButton = isNavigationButton;
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@nativescript/core/ui/page');

var _c0 = ["*"];
var ɵ0 = function (parent, child, next) {
    if (element_registry_1.isInvisibleNode(child)) {
        return;
    }
    else if (isNavigationButton(child)) {
        parent.navigationButton = child;
        child.parentNode = parent;
    }
    else if (isActionItem(child)) {
        addActionItem(parent, child, next);
        child.parentNode = parent;
    }
    else if (element_registry_1.isView(child)) {
        parent.titleView = child;
    }
}, ɵ1 = function (parent, child) {
    if (element_registry_1.isInvisibleNode(child)) {
        return;
    }
    else if (isNavigationButton(child)) {
        if (parent.navigationButton === child) {
            parent.navigationButton = null;
        }
        child.parentNode = null;
    }
    else if (isActionItem(child)) {
        parent.actionItems.removeItem(child);
        child.parentNode = null;
    }
    else if (element_registry_1.isView(child) && parent.titleView && parent.titleView === child) {
        parent.titleView = null;
    }
};
exports.ɵ0 = ɵ0;
exports.ɵ1 = ɵ1;
var actionBarMeta = {
    skipAddToDom: true,
    insertChild: ɵ0,
    removeChild: ɵ1,
};
var addActionItem = function (bar, item, next) {
    if (next) {
        insertActionItemBefore(bar, item, next);
    }
    else {
        appendActionItem(bar, item);
    }
};
var ɵ2 = addActionItem;
exports.ɵ2 = ɵ2;
var insertActionItemBefore = function (bar, item, next) {
    var actionItems = bar.actionItems;
    var actionItemsCollection = actionItems.getItems();
    var indexToInsert = actionItemsCollection.indexOf(next);
    actionItemsCollection.splice(indexToInsert, 0, item);
    actionItems.setItems(actionItemsCollection);
};
var ɵ3 = insertActionItemBefore;
exports.ɵ3 = ɵ3;
var appendActionItem = function (bar, item) {
    bar.actionItems.addItem(item);
};
var ɵ4 = appendActionItem;
exports.ɵ4 = ɵ4;
element_registry_1.registerElement("ActionBar", function () { return action_bar_1.ActionBar; }, actionBarMeta);
element_registry_1.registerElement("ActionItem", function () { return action_bar_1.ActionItem; });
element_registry_1.registerElement("NavigationButton", function () { return action_bar_1.NavigationButton; });
var ActionBarComponent = /** @class */ (function () {
    function ActionBarComponent(element, page) {
        this.element = element;
        this.page = page;
        if (!this.page) {
            throw new Error("Inside ActionBarComponent but no Page found in DI.");
        }
        if (lang_facade_1.isBlank(this.page.actionBarHidden)) {
            this.page.actionBarHidden = false;
        }
        this.page.actionBar = this.element.nativeElement;
        this.page.actionBar.update();
    }
    ActionBarComponent = __decorate([ __metadata("design:paramtypes", [core_1.ElementRef, page_1.Page])
    ], ActionBarComponent);
ActionBarComponent.ɵfac = function ActionBarComponent_Factory(t) { return new (t || ActionBarComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Page)); };
ActionBarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ActionBarComponent, selectors: [["ActionBar"]], ngContentSelectors: _c0, decls: 1, vars: 0, template: function ActionBarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ActionBarComponent, [{
        type: core_1.Component,
        args: [{
                selector: "ActionBar",
                template: "<ng-content></ng-content>"
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.Page }]; }, null); })();
    return ActionBarComponent;
}());
exports.ActionBarComponent = ActionBarComponent;
var ActionBarScope = /** @class */ (function () {
    function ActionBarScope(page) {
        this.page = page;
        if (!this.page) {
            throw new Error("Inside ActionBarScope but no Page found in DI.");
        }
    }
    ActionBarScope.prototype.onNavButtonInit = function (navBtn) {
        this.page.actionBar.navigationButton = navBtn.element.nativeElement;
    };
    ActionBarScope.prototype.onNavButtonDestroy = function (navBtn) {
        var nav = navBtn.element.nativeElement;
        if (nav && this.page.actionBar.navigationButton === nav) {
            this.page.actionBar.navigationButton = null;
        }
    };
    ActionBarScope.prototype.onActionInit = function (item) {
        this.page.actionBar.actionItems.addItem(item.element.nativeElement);
    };
    ActionBarScope.prototype.onActionDestroy = function (item) {
        if (item.element.nativeElement.actionBar) {
            this.page.actionBar.actionItems.removeItem(item.element.nativeElement);
        }
    };
    ActionBarScope = __decorate([ __metadata("design:paramtypes", [page_1.Page])
    ], ActionBarScope);
ActionBarScope.ɵfac = function ActionBarScope_Factory(t) { return new (t || ActionBarScope)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Page)); };
ActionBarScope.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ActionBarScope, selectors: [["ActionBarExtension"]], decls: 0, vars: 0, template: function ActionBarScope_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ActionBarScope, [{
        type: core_1.Component,
        args: [{
                selector: "ActionBarExtension",
                template: ""
            }]
    }], function () { return [{ type: ɵngcc1.Page }]; }, null); })();
    return ActionBarScope;
}());
exports.ActionBarScope = ActionBarScope;
var ActionItemDirective = /** @class */ (function () {
    function ActionItemDirective(element, ownerScope) {
        this.element = element;
        this.ownerScope = ownerScope;
        if (this.ownerScope) {
            this.ownerScope.onActionInit(this);
        }
    }
    ActionItemDirective.prototype.ngOnDestroy = function () {
        if (this.ownerScope) {
            this.ownerScope.onActionDestroy(this);
        }
    };
    ActionItemDirective = __decorate([ __param(1, core_1.Optional()),
        __metadata("design:paramtypes", [core_1.ElementRef, ActionBarScope])
    ], ActionItemDirective);
ActionItemDirective.ɵfac = function ActionItemDirective_Factory(t) { return new (t || ActionItemDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ActionBarScope, 8)); };
ActionItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ActionItemDirective, selectors: [["ActionItem"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ActionItemDirective, [{
        type: core_1.Directive,
        args: [{
                selector: "ActionItem" // tslint:disable-line:directive-selector
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ActionBarScope, decorators: [{
                type: core_1.Optional
            }] }]; }, null); })();
    return ActionItemDirective;
}());
exports.ActionItemDirective = ActionItemDirective;
var NavigationButtonDirective = /** @class */ (function () {
    function NavigationButtonDirective(element, ownerScope) {
        this.element = element;
        this.ownerScope = ownerScope;
        if (this.ownerScope) {
            this.ownerScope.onNavButtonInit(this);
        }
    }
    NavigationButtonDirective.prototype.ngOnDestroy = function () {
        if (this.ownerScope) {
            this.ownerScope.onNavButtonDestroy(this);
        }
    };
    NavigationButtonDirective = __decorate([ __param(1, core_1.Optional()),
        __metadata("design:paramtypes", [core_1.ElementRef, ActionBarScope])
    ], NavigationButtonDirective);
NavigationButtonDirective.ɵfac = function NavigationButtonDirective_Factory(t) { return new (t || NavigationButtonDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ActionBarScope, 8)); };
NavigationButtonDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NavigationButtonDirective, selectors: [["NavigationButton"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NavigationButtonDirective, [{
        type: core_1.Directive,
        args: [{
                selector: "NavigationButton" // tslint:disable-line:directive-selector
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ActionBarScope, decorators: [{
                type: core_1.Optional
            }] }]; }, null); })();
    return NavigationButtonDirective;
}());
exports.NavigationButtonDirective = NavigationButtonDirective;
